x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_view/category_stopped_partitions.tsx

```diff
@@ -0,0 +1,121 @@
+ */
+
+import React, { FC, useContext, useEffect, useState, useMemo } from 'react';
+import { EuiBasicTable, EuiCallOut, EuiSpacer, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { Subscription } from 'rxjs';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { CategorizationJobCreator } from '../../../../../common/job_creator';
+import { Results } from '../../../../../common/results_loader';
+import { ml } from '../../../../../../../services/ml_api_service';
+
+const NUMBER_OF_PREVIEW = 5;
+export const CategoryStoppedPartitions: FC = () => {
+  const { jobCreator: jc, resultsLoader } = useContext(JobCreatorContext);
+  const jobCreator = jc as CategorizationJobCreator;
+  const [tableRow, setTableRow] = useState<Array<{ partitionName: string }>>([]);
+  const [hasStoppedPartitions, setHasStoppedPartitions] = useState(false);
+
+  let _resultsSubscription: undefined | Subscription;
+
+  const columns = useMemo(
+    () => [
+      {
+        field: 'partitionName',
+        name: i18n.translate(
+          'xpack.ml.newJob.wizard.pickFieldsStep.stoppedPartitionsPreviewColumnName',
+          {
+            defaultMessage: 'Stopped partition {count, plural, one {name} other {names}}',
+            values: { count: tableRow.length },
+          }
+        ),
+        render: (partition: any) => (
+          <EuiText size="s">
+            <code>{partition}</code>
+          </EuiText>
+        ),
+      },
+    ],
+    []
+  );
+  function setResultsWrapper(results: Results) {
+    loadCategoryStoppedPartitions();
+  }
+
+  async function loadCategoryStoppedPartitions() {
+    const results = await ml.results.getCategoryStoppedPartitions([jobCreator.jobId]);
+
+    if (
+      results?.jobs !== undefined &&
+      !Array.isArray(results?.jobs) && // if jobs is object of jobId: [partitions]
+      Array.isArray(results?.jobs[jobCreator.jobId]) &&
+      results.jobs[jobCreator.jobId].length > 0
+    ) {
+      const stoppedPartitionsPreview = results.jobs[jobCreator.jobId];
+      // once we have reached number of stopped partitions we wanted to show as preview
+      // no need to keep fetching anymore
+      if (
+        stoppedPartitionsPreview.length >= NUMBER_OF_PREVIEW &&
+        _resultsSubscription !== undefined
+      ) {
+        _resultsSubscription.unsubscribe();
+        _resultsSubscription = undefined;
+      }
+      setHasStoppedPartitions(true);
+      setTableRow(
+        stoppedPartitionsPreview.slice(0, NUMBER_OF_PREVIEW).map((partitionName) => ({
+          partitionName,
+        }))
+      );
+    }
+  }
+
+  useEffect(() => {
+    // only need to run this check if jobCreator.perPartitionStopOnWarn is turned on
+    if (jobCreator.perPartitionCategorization && jobCreator.perPartitionStopOnWarn) {
+      // subscribe to result updates
+      _resultsSubscription = resultsLoader.subscribeToResults(setResultsWrapper);
```

Comment:
keeping the subscription stored in `_resultsSubscription` which is not in a hook might will mean it is redeclared on every render.

`loadCategoryStoppedPartitions` could be moved inside this `useEffect` or `hasStoppedPartitions` could be changed to `stoppedPartitionsCount` which blocks the running of the check, rather than unsubscribing early.

Reply:
I ended up using a ref to make sure `_resultsSubscription` is not re-declared every time the component renders.